app/src/org/commcare/activities/LoginActivityUiController.java

```diff
@@ -550,6 +550,7 @@ protected void setConnectIdLoginState(boolean useConnectId) {
             text = Localization.get("login.button");
         }
         loginButton.setText(text);
+        connectLoginButton.setText(activity.getString(R.string.connect_button_logged_in));
```

Comment:
This is set statically in the XML, does it need to be updated here?

Reply:
I included these lines because when a user changes the language from the system settings while in the app, the strings don't translate correctly when they return to the app. That's why I added this code.

Reply:
Ah, yeah I see it happening too, not sure why. Add a comment above explaining why we're doing it just so we don't forget.

Reply:
Okay